actionview/test/template/tag_helper_test.rb

```diff
@@ -391,6 +391,42 @@ def test_escape_once
     assert_equal " &#X27; &#x27; &#x03BB; &#X03bb; &quot; &#39; &lt; &gt; ", escape_once(" &#X27; &#x27; &#x03BB; &#X03bb; \" ' < > ")
   end
 
+  def test_tag_attributes_inlines_html_attributes
+    expected_output = <<~HTML.strip
+      <input type="text" name="name" aria-hidden="false" aria-label="label" data-input-value="data" required="required">
+    HTML
+
+    assert_equal expected_output, render_erb(<<~HTML.strip)
+      <input type="text" <%= tag_attributes value: nil, name: "name", "aria-hidden": false, aria: { label: "label" }, data: { input_value: "data" }, required: true %>>
+    HTML
+  end
+
+  def test_tag_attributes_escapes_values
+    assert_not_includes render_erb(<<~HTML.strip), "<script>alert()</script>"
+      <input type="text" <%= tag_attributes xss: '"><script>alert()</script>' %>>
```

Comment:
I'm not totally sure if this is an XSS vector, but since we get the escaping behavior for free by using `TagBuilder#tag_options` it felt appropriate to cover it.